proposals/0030/README.md

```diff
@@ -0,0 +1,32 @@
+---
+id: 0030
+title: Configurable default CAPT templates and simplification of template overriding
+status: discussion
+authors: Chris Doherty <chris.doherty4@gmail.com>
+---
+
+# Configurable default CAPT templates and simplification of template overriding
+
+## Context
+
+When provisioning Kubernetes clusters with CAPT, CAPT will, by default, generate a Tinkerbell `Template` and submit it in conjunction with a Workflow to provision a node. 
+It offers a template override feature that can be used to specify a template in full in place of the default template.
+
+## Proposal
+
+Firstly, allow operators to specify a default template by creating it in the cluster and referencing it CAPT configuration on launch or in a `ConfigMap`. 
+This gives operators the ability to customize the default template that has to otherwise make assumptions on what to do when no template override is specified.
+
+Secondly, replace the template override feature on `TinkerbellMachineTemplate`s with a reference field to a template. 
+Using a reference field simplifies the CAPT API and enables re-use of templates when provisioning machines.
+
+## Rationale
+
+The default template generated by CAPT makes assumptions on how nodes are to be provisioned. 
+It assumes images reside in an OCI registry, the operator does not wish to adjust cloud-init configuration, nodes are to be bootstrapped using cloud-init and that users wish to use kexec to launch the node bootstrap process. 
+Allowing operators to specify a default gives operators better control over how nodes are provisioned without needing to specify overrides.
+
+Template overrides are specified using a string field on the `TinkerbellMachineTemplate`. 
+The string field makes it difficult to understand the structure of the template and requires operators to understand how it's rendered to be confident the template is correct. 
```

Comment:
Does this suggest there would be a benefit to having both the template reference and the rendered template in the resource?

Reply:
The rendered template would feature on the `Workflow` status (as it does today). I can't think of a reason to store it on the `TinkerbellMachine` if we don't need to.